website/src/preferences-context.tsx

```diff
@@ -15,51 +15,89 @@ export const usePreferences = () => useContext(PreferencesContext)
 
 export const PreferencesProvider = (props: any) => {
   // Some preferences hooks setup
-  const [viewMode, setViewMode] = useState(savedViewMode())
+  const [levelOfDetail, setLevelOfDetail] = useState(savedLevelOfDetail())
   const [cherokeeRepresentation, setCherokeeRepresentation] = useState(
     savedCherokeeRepresentation()
   )
+
+  useEffect(() => {
+    save(PreferenceKey.LevelOfDetail, levelOfDetail)
+    save(PreferenceKey.CherokeeRepresentation, cherokeeRepresentation)
+    // Remove all deprecated cookies, keep this code for at least several months
+    // to let users clear their site cookies.
+    Cookies.remove(DeprecatedCookie.LevelOfDetail)
+    Cookies.remove(DeprecatedCookie.Romanization)
+  }, [levelOfDetail, cherokeeRepresentation])
```

Comment:
It seems unnecessary to include an attempt to remove deprecated cookies everytime `levelOfDetail` or `cherokeeRepresentation`, since after its done once we shouldn't really need it done multiple times. I don't know if it involves enough performance to warrant pulling out, but I think it would be sufficient to just pull it into its own `useEffect` that only runs on first render.

Reply:
Yeah that's fair, will do.